trustmanager/keyfilestore.go

```diff
@@ -38,10 +49,96 @@ func NewKeyFileStore(baseDir string, passphraseRetriever passphrase.Retriever) (
 		return nil, err
 	}
 	cachedKeys := make(map[string]*cachedKey)
+	keyInfoMap := make(keyInfoMap)
 
-	return &KeyFileStore{SimpleFileStore: *fileStore,
+	keyStore := &KeyFileStore{SimpleFileStore: *fileStore,
 		Retriever:  passphraseRetriever,
-		cachedKeys: cachedKeys}, nil
+		cachedKeys: cachedKeys,
+		keyInfoMap: keyInfoMap,
+	}
+
+	// Load this keystore's ID --> gun/role map
+	keyStore.loadKeyInfo()
+	return keyStore, nil
+}
+
+func generateKeyInfoMap(s LimitedFileStore) map[string]KeyInfo {
+	keyInfoMap := make(map[string]KeyInfo)
+	for _, keyPath := range s.ListFiles() {
+		// Remove the prefix of the directory from the filename for GUN/role/ID parsing
+		var keyIDAndGun, keyRole string
+		if strings.HasPrefix(keyPath, notary.RootKeysSubdir+"/") {
+			keyIDAndGun = strings.TrimPrefix(keyPath, notary.RootKeysSubdir+"/")
+		} else {
+			keyIDAndGun = strings.TrimPrefix(keyPath, notary.NonRootKeysSubdir+"/")
+		}
+
+		// Separate the ID and GUN (can be empty) from the filepath
+		keyIDAndGun = strings.TrimSpace(keyIDAndGun)
+		keyGun := getGunFromFullID(keyIDAndGun)
+		keyID := filepath.Base(keyIDAndGun)
+
+		// If the key does not have a _, we'll attempt to
```

Comment:
Sorry, this comment should be clearer -- this is for reading the role from the PEM headers vs. legacy filenames